apps/desktop/src/autofill/modal/credentials/fido2-vault.component.spec.ts

```diff
@@ -0,0 +1,203 @@
+      schemas: [NO_ERRORS_SCHEMA],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(Fido2VaultComponent);
+    component = fixture.componentInstance;
+  });
+
+  function createMockCiphers(): CipherView[] {
+    const cipher1 = new CipherView();
+    cipher1.id = "cipher-1";
+    cipher1.name = "Test Cipher 1";
+    cipher1.type = CipherType.Login;
+    cipher1.login = new LoginView();
+    cipher1.login.username = "test1@example.com";
+    cipher1.reprompt = CipherRepromptType.None;
+    cipher1.deletedDate = null;
+
+    const cipher2 = new CipherView();
+    cipher2.id = "cipher-2";
+    cipher2.name = "Test Cipher 2";
+    cipher2.type = CipherType.Login;
+    cipher2.login = new LoginView();
+    cipher2.login.username = "test2@example.com";
+    cipher2.reprompt = CipherRepromptType.None;
+    cipher2.deletedDate = null;
+
+    const cipher3 = new CipherView();
+    cipher3.id = "cipher-3";
+    cipher3.name = "Test Cipher 3";
+    cipher3.type = CipherType.Login;
+    cipher3.login = new LoginView();
+    cipher3.login.username = "test3@example.com";
+    cipher3.reprompt = CipherRepromptType.Password;
+    cipher3.deletedDate = null;
+
+    return [cipher1, cipher2, cipher3];
+  }
+
+  describe("ngOnInit", () => {
+    it("should initialize session and load ciphers successfully", async () => {
+      const mockCiphers = createMockCiphers();
+      mockCipherService.getAllDecryptedForIds.mockResolvedValue(mockCiphers);
+
+      await component.ngOnInit();
+
+      expect(mockAccountService.setShowHeader).toHaveBeenCalledWith(false);
+      expect(mockFido2UserInterfaceService.getCurrentSession).toHaveBeenCalled();
+      expect(component.session).toBe(mockSession);
+      expect(component.cipherIds$).toBe(mockSession.availableCipherIds$);
+    });
+
+    it("should handle error when no active session found", async () => {
+      mockFido2UserInterfaceService.getCurrentSession.mockReturnValue(null);
+
+      await expect(component.ngOnInit()).rejects.toThrow();
+    });
+
+    it("should filter out deleted ciphers", async () => {
+      const mockCiphers = createMockCiphers();
+      mockCiphers[1].deletedDate = new Date();
+      mockCipherService.getAllDecryptedForIds.mockResolvedValue(mockCiphers);
+
+      await component.ngOnInit();
+      await new Promise((resolve) => setTimeout(resolve, 0));
+
+      let ciphersResult: CipherView[] = [];
+      component.ciphers$.subscribe((ciphers) => {
+        ciphersResult = ciphers;
+      });
+
+      expect(ciphersResult).toHaveLength(2);
+      expect(ciphersResult.every((cipher) => !cipher.deletedDate)).toBe(true);
+    });
+  });
+
+  describe("ngOnDestroy", () => {
+    it("should restore header visibility and clean up", async () => {
+      await component.ngOnInit();
+      await component.ngOnDestroy();
+
+      expect(mockAccountService.setShowHeader).toHaveBeenCalledWith(true);
+    });
+  });
+
+  describe("chooseCipher", () => {
+    beforeEach(() => {
```

Comment:
non-blocking nit; you could assign the variable once and only re-assign the `reprompt` property in each test block 🤷 

```suggestion
    let cipher = createMockCiphers()[0];

    beforeEach(() => {
```